test/integration/controller/jobs/jobset/jobset_controller_test.go

```diff
@@ -1051,3 +1051,132 @@ var _ = ginkgo.Describe("JobSet controller interacting with scheduler", ginkgo.O
+		gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+		onDemandFlavor = testing.MakeResourceFlavor("on-demand").Label(instanceKey, "on-demand").Obj()
+		gomega.Expect(k8sClient.Create(ctx, onDemandFlavor)).Should(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, onDemandFlavor, true)
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, localQueue, true)
+	})
+
+	ginkgo.It("Should finish the preemption when the jobset becomes inactive", func() {
+		jobSet := testingjobset.MakeJobSet(jobSetName, ns.Name).ReplicatedJobs(
+			testingjobset.ReplicatedJobRequirements{
+				Name:        "replicated-job-1",
+				Replicas:    1,
+				Parallelism: 1,
+				Completions: 1,
+			},
+		).Queue(localQueue.Name).
+			Suspend(false).
+			Obj()
+		createdWorkload := &kueue.Workload{}
+		var wlLookupKey types.NamespacedName
+
+		ginkgo.By("create the jobset and admit the workload", func() {
+			gomega.Expect(k8sClient.Create(ctx, jobSet)).To(gomega.Succeed())
+			wlLookupKey = types.NamespacedName{Name: workloadjobset.GetWorkloadNameForJobSet(jobSet.Name, jobSet.UID), Namespace: ns.Name}
+			gomega.Eventually(func() error { return k8sClient.Get(ctx, wlLookupKey, createdWorkload) }, util.Timeout, util.Interval).Should(gomega.Succeed())
+			admission := testing.MakeAdmission(localQueue.Name).PodSets(
+				kueue.PodSetAssignment{
+					Name: createdWorkload.Spec.PodSets[0].Name,
+					Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+						corev1.ResourceCPU: kueue.ResourceFlavorReference(onDemandFlavor.Name),
+					},
+				},
+			).Obj()
+			gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, createdWorkload, admission)).To(gomega.Succeed())
+			util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
+		})
+
+		ginkgo.By("wait for the jobset to be unsuspended", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: jobSetName, Namespace: ns.Name}, jobSet)).Should(gomega.Succeed())
+				g.Expect(ptr.Deref(jobSet.Spec.Suspend, false)).Should(gomega.BeTrue())
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("mark the jobset as active", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(jobSet), jobSet)).To(gomega.Succeed())
+				jobSet.Status.ReplicatedJobsStatus = make([]jobsetapi.ReplicatedJobStatus, 1)
+				jobSet.Status.ReplicatedJobsStatus[0].Active = 1
+				jobSet.Status.ReplicatedJobsStatus[0].Name = jobSet.Spec.ReplicatedJobs[0].Name
+				g.Expect(k8sClient.Status().Update(ctx, jobSet)).To(gomega.Succeed())
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("preempt the workload", func() {
+			gomega.Eventually(func() error {
```

Comment:
As @trasc mentioned in another review comment, could you use the nested gomega instance approach for more debuggability?